### ISSUE TITLE
`Sources` referenced, but doesn't exist in `pants.engine.target`

### ISSUE THREAD

**Issue by thejcannon** [pantsbuild/example-plugin#33]
Reference: https://github.com/pantsbuild/example-plugin/blob/1b955ede47ac7fa3a07320e495f42c6182bae788/pants-plugins/examples/bash/target_types.py#L15

Should it be `SourcesField`?

**Comment by kaos** [comment]
Yes

**Comment by kaos** [comment]
Eg, no, sorry, it's MultipleSourcesField

**Comment by kaos** [comment]
And SingleSourceField.. there's two different uses in there..

**Comment by thejcannon** [comment]
Yeah, I realized after filing this issue that this code needs some TLC <3

**Comment by Eric-Arellano** [comment]
Yeah, it needs major TLC....I haven't had much time to update it and also know that we probably want to start it all from scratch a la https://github.com/pantsbuild/pants/issues/11825.
